crates/noirc_frontend/src/ast/statement.rs

```diff
@@ -65,6 +71,18 @@ impl Statement {
                 self
             }
 
+            Statement::Return { expr, semi: false } => {
+                if !last_statement_in_block && semi.is_none() {
+                    let reason = "Expected a ; separating these two statements".to_string();
+                    emit_error(ParserError::with_reason(reason, span));
```

Comment:
As of https://github.com/noir-lang/noir/pull/1286, `ParserError::with_reason` takes an enum (hence the failing tests). You can add a variant in `parser/errors.rs`.